Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1

```diff
@@ -171,13 +171,68 @@ param (
         Write-Verbose "Unable to get the path for net.exe. Net command will be 'net'"
         return 'net'
     }
+
+    function Create-DestinationDirectory(
+        [string]$path
+    )
+    {
+        $psCredentialObject = New-Object PSCredential($credential.UserName, (ConvertTo-SecureString $credential.Password -AsPlainText -Force))
+        $destPath = $path
+        $foundParentPath = $false
+        $isRoot = $false
+
+        Write-Verbose "Creating path to directory: $path"
+        while($destPath -and (-not $foundParentPath))
+        {
+            try
+            {
+                New-PSDrive -Name WFCPSDrive -PSProvider FileSystem -Root $destPath -Credential $psCredentialObject
+                $foundParentPath = $true
+                Write-Verbose "Found parent path"
+                $relativePath = $path.Substring($destPath.Length)
+                New-Item -ItemType Directory WFCPSDrive:$relativePath -Force
+                Write-Verbose "Created directory"
```

Comment:
is there any lag to propagate the new directory created on network path info? what I mean for e.g dest is \\share\a\b\c\d\e\f . Current logic is to create \\share\a then \\share\a\b then \\share\a\b\c..but what if thought \\share\a got created in next loop for creating \\share\a\b, we are not able to get \\share\a and tries to recreate same again

Reply:
As discussed offline, we are creating all the required folders in one step.